### DISCUSSION TITLE
Feature Suggestion: Add a “Go to Folder” Option for Direct Navigation

### DISCUSSION THREAD

**Discussion by josepacelli** [iterate-ch/cyberduck#16684] [Ideas]
#### **Description**  
Implement a feature that allows users to **navigate directly to a specific folder** by entering its full path. This functionality would be particularly useful for environments with deep or complex directory structures, eliminating the need for manual browsing.

#### **Proposed Features**  

1. **Direct Path Input Field**  
   - Add an input field (e.g., in the toolbar or navigation menu) where users can type or paste the **full folder path**.  
   - Example: `/opt/log/2024/12/16/09/`.

2. **Keyboard Shortcut**  
   - Introduce a keyboard shortcut (e.g., `Ctrl+G` on Windows or `Cmd+G` on macOS) to open a dialog box for entering the folder path.

3. **Path Validation**  
   - When the user presses "Enter" after entering the path, the Cyberduck client validates the folder's existence and navigates directly to it.  
   - If the path is invalid or does not exist, display a user-friendly error message.

4. **Recent Folder History**  
   - Include a history of recently accessed folder paths for quick and recurring navigation.  

#### **Benefits**  
- **Improved Productivity**: Reduces the time spent manually navigating through complex directory structures.  
- **Efficiency**: Allows quick and direct access to specific folders, improving workflow for advanced users.  
- **User-Friendly**: Simplifies navigation for users who already know the folder paths.  

#### **Target Users**  
- System administrators who need to access logs or specific directories quickly.  
- Developers and IT professionals working with deeply nested file structures on remote servers.  
- Users of FTP/SFTP, WebDAV, and cloud storage services like Amazon S3.  

#### **Example Use Case**  
- A user presses `Cmd+G` (Mac) or `Ctrl+G` (Windows) and inputs the path: `/opt/log/2024/12/16/09/`. Cyberduck validates the path and immediately opens the specified folder.

**Comment by dkocher** [comment]
Did you try _Go → Go to Folder… ⌘G_.
